Public/Src/FrontEnd/Nuget/NugetResolver.cs

```diff
@@ -62,12 +69,87 @@ public override async Task<bool> InitResolverAsync(IResolverSettings resolverSet
 
             m_owningModules = new Dictionary<ModuleId, Package>();
 
-            foreach (var package in maybePackages.Result)
+            foreach (var package in maybePackages.Result.Values.SelectMany(v => v))
             {
                 m_packages[package.Id] = package;
                 m_owningModules[package.ModuleId] = package;
             }
 
+            if (Configuration.FrontEnd.GenerateCgManifestForNugets.IsValid ||
+                Configuration.FrontEnd.ValidateCgManifestForNugets.IsValid)
+            {
+                var cgManfiestGenerator = new NugetCgManifestGenerator(Context);
+                string generatedCgManifest = cgManfiestGenerator.GenerateCgManifestForPackages(maybePackages.Result);
+                string existingCgManifest = "{}";
+
+                if ( !Configuration.FrontEnd.GenerateCgManifestForNugets.IsValid &&
+                      Configuration.FrontEnd.ValidateCgManifestForNugets.IsValid )
+                {
+                    // Validation of existing cgmainfest.json results in failure due to mismatch. Should fail the build in this case.
+                    try
+                    {
+                        existingCgManifest = File.ReadAllText(Configuration.FrontEnd.ValidateCgManifestForNugets.ToString(Context.PathTable));
+                        FrontEndHost.Engine.RecordFrontEndFile(
+                            Configuration.FrontEnd.ValidateCgManifestForNugets,
+                            NugetResolverName);
+                    }
+                    // CgManifest FileNotFound, log error and fail build
+                    catch (DirectoryNotFoundException e)
+                    {
+                        throw new BuildXLException(@"Cannot read cgmanifest.json file from disk", e);
+                    }
+                    catch (FileNotFoundException e)
+                    {
+                        throw new BuildXLException(@"Cannot read cgmanifest.json file from disk", e);
+                    }
+                    if (!cgManfiestGenerator.CompareForEquality(generatedCgManifest, existingCgManifest))
+                    {
+                        throw new BuildXLException(@"Existing cgmanifest.json file is outdated, please generate a new one using /GenerateCgManifestForNugets:path");
+                    }

```

Comment:
Is the caller handling the error logging? Our typical pattern would be to log a unique error event here and then return false to indicate teh failure for the caller